python/gtsam/examples/TranslationAveragingExample.py

```diff
@@ -0,0 +1,122 @@
+"""
+GTSAM Copyright 2010-2018, Georgia Tech Research Corporation,
+Atlanta, Georgia 30332-0415
+All Rights Reserved
+Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+See LICENSE for the license information
+
+This example shows how 1dsfm uses outlier rejection (MFAS) and optimization (translation recovery)
+together for estimating global translations from relative translation directions and global rotations.
+The purpose of this example is to illustrate the connection between these two classes using a small SfM dataset.
+
+Author: Akshay Krishnan
+Date: September 2020
+"""
+
+from collections import defaultdict
+from typing import Tuple, List
+
+import numpy as np
+
+import gtsam
+from gtsam.examples import SFMdata
+
+
+def get_data() -> Tuple[gtsam.Values, List[gtsam.BinaryMeasurementUnit3]]:
+    """"Returns data from SfMData.createPoses(). This contains global rotations and unit translations directions."""
+    # Using toy dataset in SfMdata for example.
+    poses = SFMdata.createPoses(gtsam.Cal3_S2(50.0, 50.0, 0.0, 50.0, 50.0))
+    # Rotations of the cameras in the world frame - wRc.
+    rotations = gtsam.Values()
+    # Normalized translation directions for pairs of cameras - from first camera to second,
+    # in the coordinate frame of the first camera.
+    translation_directions = []
+    for i in range(0, len(poses) - 2):
+        # Add the rotation.
+        rotations.insert(i, poses[i].rotation())
+        # Create unit translation measurements with next two poses.
+        for j in range(i + 1, i + 3):
+            i_Z_j = gtsam.Unit3(poses[i].rotation().unrotate(
```

Comment:
i think this line is not super readable. can we break it up by semantics, or add inline comments to explain it?

I think it will probably be easier to parse once I can figure out which way `poses[j]` represents.